server/stream.go

```diff
@@ -2686,9 +2686,13 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 			return fmt.Errorf("last msgid mismatch: %q vs %q", lmsgId, last)
 		}
 		// Expected last sequence per subject.
-		if seq := getExpectedLastSeqPerSubject(hdr); seq > 0 {
+		if seq, exists := getExpectedLastSeqPerSubject(hdr); exists {
 			// TODO(dlc) - We could make a new store func that does this all in one.
 			_, lseq, _, _, _, err := mset.store.LoadLastMsg(subject)
+			// If seq passed in is zero that signals we expect no msg to be present.
```

Comment:
I think that this change may break MQTT code that uses this header:
```
=== RUN   TestMQTTCluster
=== RUN   TestMQTTCluster/first_start
=== RUN   TestMQTTCluster/first_start/qos_0
    mqtt_test.go:2755: Error reading: EOF
--- FAIL: TestMQTTCluster (7.10s)
    --- FAIL: TestMQTTCluster/first_start (6.68s)
        --- FAIL: TestMQTTCluster/first_start/qos_0 (1.13s)
```
I set this header and initially seq is 0, however, it does not mean that I was not expecting any message to be set. Of course, we can fix the MQTT code and not set the header if know seq is 0.

Reply:
Actually, no, I don't think MQTT code should change. If I don't set the expected to 0, then I have the issue where 2 or more may be updating at the same time, and this is wrong.

So I think overloading the meaning of this header should be rethought.

Reply:
Humm... I guess original code would have ignored it anyway, so for sure we could fix the MQTT code and/or mqtt test..

Reply:
We paired and fixed.